include/zephyr/arch/exception.h

```diff
@@ -0,0 +1,37 @@
+/* exception.h - automatically selects the correct exception.h file to include */
+
+/*
+ * Copyright (c) 2024 Meta Platforms
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_ARCH_EXCEPTION_H_
+#define ZEPHYR_INCLUDE_ARCH_EXCEPTION_H_
+
+#if defined(CONFIG_X86)
+#include <zephyr/arch/x86/ia32/exception.h>
+#elif defined(CONFIG_X86_64)
+#include <zephyr/arch/x86/intel64/exception.h>
+#elif defined(CONFIG_ARM64)
+#include <zephyr/arch/arm64/exception.h>
+#elif defined(CONFIG_ARM)
+#include <zephyr/arch/arm/exception.h>
+#elif defined(CONFIG_ARC)
+#include <zephyr/arch/arc/v2/exception.h>
+#elif defined(CONFIG_NIOS2)
+#include <zephyr/arch/nios2/exception.h>
+#elif defined(CONFIG_RISCV)
+#include <zephyr/arch/riscv/exception.h>
+#elif defined(CONFIG_XTENSA)
+#include <zephyr/arch/xtensa/exception.h>
+#elif defined(CONFIG_MIPS)
+#include <zephyr/arch/mips/exception.h>
+#elif defined(CONFIG_ARCH_POSIX)
+#include <zephyr/arch/posix/exception.h>
+#elif defined(CONFIG_SPARC)
+#include <zephyr/arch/sparc/exception.h>
+#else
+#error "Unknown Architecture"
```

Comment:
I wouldnt error out, as it would fail for out of tree architectures.
CC @daor-oti

Reply:
Just want to point out that this isn't something new:

https://github.com/zephyrproject-rtos/zephyr/blob/127cb9edb6c6594e2f09dea7efb335db3cb1f43a/include/zephyr/arch/arch_inlines.h#L35-L37

https://github.com/zephyrproject-rtos/zephyr/blob/127cb9edb6c6594e2f09dea7efb335db3cb1f43a/include/zephyr/arch/cpu.h#L34-L36

Another PR is required if normalization is required

Reply:
We support out of tree architectures, SOCs, and boards. For out of tree architectures, because they are so rare, out of tree support gets a bit broken every now and then with this kind of changes. This can be worked around/hacked out of tree. But that does not mean we should just keep making it worse for no reason.
I don't think there is any benefit to add this check and error here. So I'd still ask you to not add it.

Reply:
Fair enough, added 2 more commits that will make these headers more consistent